lib/AST/NameLookup.cpp

```diff
@@ -2319,6 +2319,7 @@ directReferencesForTypeRepr(Evaluator &evaluator,
 
   case TypeReprKind::OpaqueReturn:
   case TypeReprKind::NamedOpaqueReturn:
+  case TypeReprKind::Existential:
     return { };
 
```

Comment:
Should this not recurse into the constraint repr?

Reply:
I believe so, but today it probably doesn't matter since we don't allow writing things like `extension any P` or any other construct that uses this code path.